includes/Modules/Analytics_4.php

```diff
@@ -826,4 +858,447 @@ public function check_service_entity_access() {
 		return true;
 	}
 
+	/**
+	 * Creates and executes a new Analytics 4 report request.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Data_Request $data Data request object.
+	 * @return RequestInterface|WP_Error Request object on success, or WP_Error on failure.
+	 */
+	protected function create_report_request( Data_Request $data ) {
+		$request_args = array();
+
+		$option = $this->get_settings()->get();
+
+		if ( empty( $option['propertyID'] ) ) {
+			return new WP_Error( 'missing_required_setting', __( 'No connected Google Analytics 4 property ID.', 'google-site-kit' ), array( 'status' => 400 ) );
```

Comment:
A bit of a nitpick here but a 400 response would indicate that the request was malformed in some way that it should not be retried. Since this condition is based on internal state, I think a 500 error would be more appropriate, and this probably makes sense to go after the 4xx errors (i.e. once we determine the request was made correctly).

Reply:
Good spot! I've done this, and also updated the metric/dimension validation errors to return a 400 rather than the current default of 500. Also updated the tests to check for the status code.

Reply:
Thanks! As mentioned in https://github.com/google/site-kit-wp/pull/6379#discussion_r1070000992 I don't think we should change the status of the errors for the DS-related validation because that is also dependent on internal state rather than an improperly made request. I've gone ahead and reverted this part.